provider/pkg/provider/common/inputs.go

```diff
@@ -21,10 +21,15 @@ func (c *ResourceInputs) Annotate(a infer.Annotator) {
 trigger values can be of any type. If a value is different in the current update compared to the
 previous update, the resource will be replaced, i.e., the "create" command will be re-run.
 Please see the resource documentation for examples.`)
-	a.Describe(&c.Create, "The command to run on create.")
-	a.Describe(&c.Delete, `The command to run on delete. The environment variables PULUMI_COMMAND_STDOUT
-and PULUMI_COMMAND_STDERR are set to the stdout and stderr properties of the
-Command resource from previous create or update steps.`)
+	a.Describe(&c.Create, "The command to run once on resource creation.\n\n"+
+		"If an `update` command isn't provided, then `create` will also be run when the resource's inputs are modified.\n\n"+
+		"Note that this command will not be executed if the resource has already been created and its inputs are unchanged.\n\n"+
+		"Use `local.runOutput` if you need to run a command on every execution of your program.")
+	a.Describe(&c.Delete, "The command to run when the resource is updated.\n\n"+
+		"If empty, the create command will be executed instead.\n\n"+
+		"Note that this command will not run if the resource's inputs are unchanged.\n\n"+
+		"Use `local.runOutput` if you need to run a command on every execution of your program.\n\n"+
+		"The environment variables `PULUMI_COMMAND_STDOUT` and `PULUMI_COMMAND_STDERR` are set to the `stdout` and `stderr` properties of the Command resource from previous create or update steps.")
 	a.Describe(&c.Update, `The command to run on update, if empty, create will 
```

Comment:
perhaps you also meant to update the description for update 

```suggestion
	a.Describe(&c.Update, "The command to run when the resource is updated.\n\n"+
		"If empty, the create command will be executed instead.\n\n"+
		"Note that this command will not run if the resource's inputs are unchanged.\n\n"+
		"Use `local.runOutput` if you need to run a command on every execution of your program.\n\n"+
		"The environment variables `PULUMI_COMMAND_STDOUT` and `PULUMI_COMMAND_STDERR` are set to the `stdout` and `stderr` properties of the Command resource from previous create or update steps.")
```